pkg/netstore/netstore.go

```diff
@@ -78,6 +117,16 @@ func (s *store) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chun
 		if !s.validator.Validate(ch) {
 			return nil, storage.ErrInvalidChunk
 		}
+
+		// if the chunk is expected as part of the repair process, then deliver it
+		s.chunkChanMu.Lock()
```

Comment:
Yeah.. it is validated as SOC in the previous step...

Reply:
i understand what you say. But it is hard to execute this only in the case of SOC chunk without changing the validator interface? i am not sure about changing that interface. @janos WDYT?

The only other way to find it as a SOC chunks is to check the payload length. But it is not guaranteed to pass for all SOC chunks.

Reply:
We already can use soc.Validator.Validate function. There is no need to extend he interface, just use the specific validator that already exists.

Reply:
@janos That validator is called in the above few lines and the control comes down only if the SOC chunk is valid.

Reply:
I know, the validator is returning true if a chunk is valid of any type. I am saying that we already have a way to check if a chunk is valid for a specific type without changing the interface in this pr.

If you like to introduce the validator change in this PR I think that it will be out of scope. I would propose to create a PR where a validator would work in a way to return which types a chunk is identified (validated) as.

Reply:
Even though i think that adding a SOC validate will be more costlier than checking the `chunkChanMap` for every chunk, i accept this comment and added a SOC validate. I also added a comment there saying that the validate should spew out the chunk type in a future PR.